product-service/src/main/java/com/extremecoder/productservice/utils/ResponseBuilder.java

```diff
@@ -0,0 +1,80 @@
+package com.extremecoder.productservice.utils;
+
+import com.extremecoder.productservice.dto.ErrorResponseDto;
+import com.extremecoder.productservice.dto.Response;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.BindingResult;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class ResponseBuilder {
+    private ResponseBuilder() {
+    }
+
+    private static List<ErrorResponseDto> getCustomError(BindingResult result) {
+        List<ErrorResponseDto> dtoList = new ArrayList<>();
+        result.getFieldErrors().forEach(fieldError -> {
```

Comment:
We could use `map()` instead of `forEach()`. In that case we did not need to create two extra variable like: `dtoList` and `errorResponseDto`.

Reply:
Resolved this, bhai. Thanks.

Reply:
Actually i was talking about something like this:
`return result.getFieldErrors().map(error -> ErrorResponseDto.builder().field(error.getField()).message(error.getDefaultMessage()).build()).collect(Collector.toList);`

Reply:
One more thing, what if the `result` variable is null here?